packages/core/android/src/main/java/com/segmentanalyticsreactnative/AnalyticsReactNativeModule.kt

```diff
@@ -98,7 +98,7 @@ class AnalyticsReactNativeModule(reactContext: ReactApplicationContext) : ReactC
       contextInfo.putString("appVersion", appVersion)
       contextInfo.putString("buildNumber", buildNumber)
       contextInfo.putString("bundleId", bundleId)
-      contextInfo.putString("deviceId", getUniqueId(config.hasKey("collectDeviceId") && config.getBoolean("collectDeviceId")))
+      contextInfo.putString("deviceId", getUniqueId(collectDeviceId))
```

Comment:
We should check the config as before to only request it when the config says so.